src/header/header.scss

```diff
@@ -0,0 +1,227 @@
+          margin-right: $iui-l;
+        }
+
+        > .iui-brand-logo > .iui-icon {
+          width: $iui-icons-default;
+          height: $iui-icons-default;
+        }
+      }
+
+      > .iui-header-button {
+        > .iui-thumbnail-area > .iui-thumbnail {
+          width: $iui-icons-default;
+          height: $iui-icons-default;
+        }
+
+        &.iui-detailed > .iui-label {
+          padding-top: $iui-baseline * 2;
+
+          > .iui-id {
+            opacity: 0;
+          }
+        }
+      }
+    }
+    
+    > .iui-right {
+      .iui-user-icon {
+        @include iui-user-icon;
+        @include iui-user-icon-size($size: $iui-icons-large, $font-size: $iui-font-size-small);
+      }
+    }
+  }
+}
+
+@mixin iui-header-buttons {
+  > .iui-application,
+  > .iui-header-button {
+    overflow: hidden;
+    transition: height $iui-speed-fast ease-out;
+
+    > .iui-label {
+      overflow: hidden;
+      text-overflow: ellipsis;
+      transition: font-size $iui-speed-fast ease-out;
+
+      > * {
+        width: 100%;
+        overflow: hidden;
+        text-overflow: ellipsis;
+        text-align: left;
+      }
+    }
+
+    > .iui-brand-logo,
+    > .iui-thumbnail-area {
+      display: flex;
+      justify-content: center;
+      align-items: center;
+      height: 100%;
+
+      > .iui-icon,
+      > .iui-thumbnail {
+        width: $iui-icons-large;
+        height: $iui-icons-large;
+        transition: width $iui-speed-fast ease-out, height $iui-speed-fast ease-out;
+      }
+    }
+  }
+
+  > .iui-application {
+    padding: 0;
+
+    > .iui-brand-logo {
+      width: $iui-l * 3 - 1;
+      flex-shrink: 0;
+    }
+
+    > .iui-label {
+      margin-right: $iui-sm;
+      transition: margin-right $iui-speed-fast ease-out, font-size $iui-speed-fast ease-out;
+    }
+  }
+
+  > .iui-header-button {
+    > .iui-thumbnail-area {
+      width: $iui-l;
+      margin-right: $iui-sm;
+
+      > .iui-thumbnail {
+        border-radius: $iui-border-radius;
+        background-position: center;
+        background-size: cover;
+        overflow: hidden;
+        position: relative;
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        @include themed {
+          background-color: t(iui-color-background-3);
+        }
+
+        > svg {
+          width: 75%;
+          height: 75%;
+          max-width: $iui-icons-default;
+          max-height: $iui-icons-default;
+          @include themed {
+            fill: t(iui-icons-color);
+          }
+        }
+
+        > .iui-picture {
+          width: 100%;
+          height: 100%;
+          position: absolute;
+          top: 0;
+          left: 0;
+          background-position: center;
+          background-size: cover;
+        }
+      }
+    }
+
+    > .iui-label {
+      display: flex;
+      flex-direction: column;
+      align-items: flex-start;
+      transition: padding-top $iui-speed-fast ease-out;
```

Comment:
Is it needed? Because `iui-label` does not have `padding-top` and this one overrides previous transition:
![image](https://user-images.githubusercontent.com/36186912/117105124-138a8480-ad86-11eb-9576-c146b58e9f83.png)